app/controllers/comment_controller.php

```diff
@@ -0,0 +1,30 @@
+<?php 
+Class CommentController extends AppController
+{
+    public function mostLiked()
+    {
+        $comments = Comment::get_most_liked();
+        $this->set(get_defined_vars());
+    }
+
+    public function delete()
+    {
+        $comment = Comment::get(Param::get('comment_id'));
+        $comment->delete();
+        redirect(url('thread/view', array('thread_id' =>$_SESSION['thread_id'])));
+    }
+
+    public function addLike()
+    {
+        $comment = Comment::get(Param::get('comment_id'));
+        $comment->addLike();
+        redirect(url('thread/view', array('thread_id' => $_SESSION['thread_id'])));
+    }
+
+    public function removeLike()
+    {
+        $comment = Comment::get(Param::get('comment_id'));
+        $comment->removeLike();
+        redirect(url('thread/view', array('thread_id' => $_SESSION['thread_id'])));   
+    }
```

Comment:
L17-L29 seems to be related, except that addLike() adds and removeLike() removes
you could combine these two methods to setLike()
and just pass a value like add/remove to know if the like would be added or removed